docs/api-testing/mark3/mocking.md

```diff
@@ -0,0 +1,271 @@
+            ],
+            "instance": "false",
+            "message": "is not of a type(s) boolean",
+            "name": "type",
+            "path": [
+                "is_admin"
+            ],
+            "property": "instance.is_admin",
+            "schema": {
+                "type": "boolean"
+            },
+            "stack": "instance.is_admin is not of a type(s) boolean"
+        }
+    ],
+    "message": "The example does not match the schema"
+}
+```
+
+The response will also contain the x-sauce-error: true header, signifying that the response is not mocked, but it's an internal error.
+
+### Validate Request
+If you want to make sure your requests are compliant with the schema, Piestry can help you.
+Run it with the `--validate-request` switch to activate the validation of inbound requests.
+
+Whenever a request is performed, it will be validated against the schema, and if a mismatch is present, an error like the following will be returned:
+
+```json
+{
+    "collection": "queryParams",
+    "errors": [
+        {
+            "argument": [
+                "integer"
+            ],
+            "instance": "aa",
+            "message": "is not of a type(s) integer",
+            "name": "type",
+            "path": [],
+            "property": "instance",
+            "schema": {
+                "type": "integer"
+            },
+            "stack": "instance is not of a type(s) integer"
+        }
+    ],
+    "message": "Wrong field types"
+}
+```
+
+The response will also contain the x-sauce-error: true header, signifying that the response is not mocked, but it's an internal error.
+
+
+### Dynamic examples
+The system allows for examples containing dynamic data using the Handlebars markup. Remember that if you use dynamic examples in your OpenAPI specs, your spec will reduce its usability for documentation purposes as documentation renderers don't support it.
+
+To have dynamic parameters, you simply place an expression between double curly brackets as in `{{requestUrl}}`.
+
+The available objects in the scope are the same as the ones used by x-sauce-cond, so: `uriParams`, `queryParams`, `headers`, `body`.
+
+As an example, the following template will echo the shape of the request back in the response:
+
+```json
+{
+ "url":"{{requestUrl}}",
+ "requestHeaders": {{json headers}},
+ "requestBody": {{json body}},
+ "ipAddress": "{{ipAddress}}"
+}
+```
+
+The "json" keyword will convert a full data structure into its JSON equivalent.
+
+
+
+
+## E2E Mode
+When Piestry is run with `--e2e`, it will turn into a reverse proxy gateway and forward the requests based to the origin, according to the OpenAPI specification. The requirement is the "server" definition of the OpenAPI spec should lead to an actual location.
+
+In this mode, you can enable contract validators as well as capture mode.
+
+### Contract Validators
+There are two types of validations you can activate, focusing on different areas.
+
```

Comment:
Add here
"### Validate Request
 If you want to make sure your requests are compliant with the schema, Piestry can help you.
 Run it with the `--validate-request` switch to activate the validation of inbound requests."